### DISCUSSION TITLE
Does it support PLSQL like debugging

### DISCUSSION THREAD

**Discussion by yangmeng66** [dbeaver/dbeaver#15752] [Q&A]
Does dbeaver have the function of debugging?

like PLSQL: 
![image](https://user-images.githubusercontent.com/51857567/157188964-212790f5-d5f8-4a9a-8f5d-824736dccfff.png)

**Comment by ShadelessFox** [comment]
Hello @yangmeng66,

DBeaver supports PL/SQL procedure debugging.

Please see https://dbeaver.com/wiki/PGDebugger

**Reply by yangmeng66** [reply]
hello,
Does this only provide debugging function for PostgreSQL??? Can other databases be used? Such as Oracle and so on
I found that this requires a postgresql12 environment

I'm confused now........


![image](https://user-images.githubusercontent.com/51857567/157833244-436f929c-83a6-4f9c-a9fd-d87556023934.png)

**Reply by ShadelessFox** [reply]
DBeaver only supports PostgreSQL PL/SQL (PL/pgSQL) debugging. Pardon me for making you confused.

**Reply by yangmeng66** [reply]
hello, 
 It is OK
Is there really no other way to debug other databases?     (for example, Oracle, or its own JDBC compliant database)

Or do you know that there are other database client tools similar to dbeaver that have the function of debugging??? (such as PLSQL)

**Reply by ShadelessFox** [reply]
Unfortunately, there's no way until we implement it.

I don't know any other database clients that suit your needs, sorry.

**Reply by yangmeng66** [reply]
hello,  why?
I installed PostgreSQL as instructed, but
sudo apt install postgresql-12-pldebugger
sudo service postgresql restart


![image](https://user-images.githubusercontent.com/51857567/157855814-d37625d4-f971-4ee9-912e-a55ee7cfed66.png)

**Reply by ShadelessFox** [reply]
I'm not familiar with setting up the PostgreSQL debugger on Linux.

Perhaps this answer on StackOverflow might help: https://stackoverflow.com/a/45090014

**Reply by yangmeng66** [reply]
hello,
It's already installed,
but how  to open this inteface
![image](https://user-images.githubusercontent.com/51857567/158102112-8b02b510-49dd-409e-af4f-7f7f14f2ce44.png)


![image](https://user-images.githubusercontent.com/51857567/158102051-c2459f85-19c5-4733-913f-a4f8d224bdca.png)

![image](https://user-images.githubusercontent.com/51857567/158117421-745e484a-3e6e-4d36-9bf0-1fd45ebb543a.png)

**Reply by yangmeng66** [reply]
@ShadelessFox @okeuday ,, hello, Can you help me solve this problem?
![image](https://user-images.githubusercontent.com/51857567/161954314-bf4e50de-d27c-49cd-8ae1-e4dd83f33aca.png)

**Reply by yangmeng66** [reply]
@ShadelessFox  what  mean it is?
NB. You must only toggle the breakpoints when Show header option is not on.
